draft-ietf-tls-esni.md

```diff
@@ -546,14 +556,14 @@ byte, and the serialized ECHConfig.
 The value of the "encrypted_client_hello" extension in the ClientHelloOuter is
 a `ClientECH` with the following values:
 
+- `config_id`, the identifier corresponding to the chosen ECHConfig structure;
 - `cipher_suite`, the client's chosen cipher suite;
-- `config_id`, the identifier of the chosen ECHConfig structure;
 - `enc`, as computed above; and
 - `payload`, as computed above.
 
-If optional configuration identifiers (see {{optional-configs}})) are used, the
-`config_id` field MAY be empty or randomly generated. Unless specified by the
-application using (D)TLS or externally configured on both sides,
+If optional configuration identifiers (see {{optional-configs}})) are used,
+`config_id` SHOULD be set to a randomly generated eight bytes. Unless specified
+by the application using (D)TLS or externally configured on both sides,
```

Comment:
So when is empty valid?

Reply:
It's always valid, I think. The server will just have to do trial decryption.

Reply:
An empty identifier is just one possible value in the space.  It doesn't mean that other configurations aren't identified by non-empty identifiers.

Reply:
What I'm saying here is that we have a way to indicate the identifier is optional, and that's to have it be 8 random bytes. So when would you use empty?

Reply:
Whenever the application wants it, I suppose? We're just recommending that these be random when optional. As Martin says, they don't need to be. Can you propose text if you think something else needs to be said here?

Reply:
I think we should forbid the empty identifier.

Reply:
I'd like to not do anything here, especially given #379, wherein we move from a vector to a `uint`. Would that be OK?